lib/view_model/contact_list/contact_list_view_model.dart

```diff
@@ -8,20 +8,35 @@ import 'package:cake_wallet/entities/contact_record.dart';
 import 'package:cake_wallet/entities/contact.dart';
 import 'package:cake_wallet/utils/mobx.dart';
 import 'package:cw_core/crypto_currency.dart';
+import 'package:collection/collection.dart';
 
 part 'contact_list_view_model.g.dart';
 
 class ContactListViewModel = ContactListViewModelBase with _$ContactListViewModel;
 
 abstract class ContactListViewModelBase with Store {
-  ContactListViewModelBase(this.contactSource, this.walletInfoSource, this._currency)
+  ContactListViewModelBase(
+      this.contactSource, this.walletInfoSource, this._currency, this.isAutoGenerateEnabled)
       : contacts = ObservableList<ContactRecord>(),
         walletContacts = [] {
     walletInfoSource.values.forEach((info) {
-      if (info.addresses?.isNotEmpty ?? false) {
-        info.addresses?.forEach((address, label) {
-          final name = label.isNotEmpty ? info.name + ' ($label)' : info.name;
+      if (info.addresses == null || info.addresses!.isEmpty) return;
 
+      if (isAutoGenerateEnabled && info.type == WalletType.monero && info.addressInfos != null) {
+        info.addressInfos!.forEach((key, value) {
+          final nextUnusedAddress = value.firstWhereOrNull(
+              (addressInfo) => !(info.usedAddresses?.contains(addressInfo.address) ?? false));
```

Comment:
shouldn't this be known by `address` since we are assigning it the `last` address in the unused addresses?